src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloudMultipeMatchingConfigurationsIT.java

```diff
@@ -0,0 +1,128 @@
+
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.DEB_JAVA_STARTUP_SCRIPT;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.LABEL;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.NULL_TEMPLATE;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.NUM_EXECUTORS;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.TEST_TIMEOUT_MULTIPLIER;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.getLabel;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.initClient;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.initCloud;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.initCredentials;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.instanceConfiguration;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.teardownResources;
+import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.Lists;
+import com.google.jenkins.plugins.computeengine.ComputeEngineCloud;
+import com.google.jenkins.plugins.computeengine.InstanceConfiguration;
+import com.google.jenkins.plugins.computeengine.client.ComputeClient;
+import hudson.model.Node;
+import hudson.model.labels.LabelAtom;
+import hudson.slaves.NodeProvisioner.PlannedNode;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.jvnet.hudson.test.JenkinsRule;
+
+/**
+ * Integration test suite for {@link ComputeEngineCloud}. Verifies that instances can be created
+ * with multiple matching {@link InstanceConfiguration} and that these instances are properly
+ * provisioned.
+ */
+public class ComputeEngineCloudMultipeMatchingConfigurationsIT {
+  private static Logger log =
+      Logger.getLogger(ComputeEngineCloudMultipeMatchingConfigurationsIT.class.getName());
+
+  private static final String DESC_1 = "type_1";
+  private static final String DESC_2 = "type_2";
+
+  @ClassRule
+  public static Timeout timeout = new Timeout(5 * TEST_TIMEOUT_MULTIPLIER, TimeUnit.MINUTES);
+
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+  private static ComputeClient client;
+  private static Map<String, String> label =
+      getLabel(ComputeEngineCloudMultipeMatchingConfigurationsIT.class);
+  private static Collection<PlannedNode> planned;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    log.info("init");
+    initCredentials(jenkinsRule);
+    ComputeEngineCloud cloud = initCloud(jenkinsRule);
+    client = initClient(jenkinsRule, label, log);
+
+    InstanceConfiguration.Builder builder1 =
```

Comment:
At current moments this builders are not builders at all..
They are just interfaces to modify instances under the hood..

It should be fixed via #84